src/bencher/db.rs

```diff
@@ -232,7 +232,12 @@ impl Task {
                 let mut value = vec![0; self.val_len];
                 random.fill_bytes(value.as_mut_slice());
                 self.db
-                    .put_with_options(key, value.as_ref(), &self.write_options)
+                    .put_with_options(
+                        key,
+                        value.as_ref(),
+                        &self.write_options,
```

Comment:
I was digesting the suggestion as well, I like the ability to add more options without making a backwards incompatible change in theory but I prefer the simplicity of the current API. OTOH, I feel like if we want to add more options they really should be part of either `PutOptions` or `WriteOptions` because we've decided that we'll have on "Options" per operation. So something like `TxnOptions` would either be specifically on transactional operations (not this top level one) or if there's a single-update transaction somehow it should be part of `PutOptions`. The example isn't great, but does the gist of that make sense?

Reply:
> I'm kind of partial to the current approach... Just seems more straight forward.
Yeah I'm fine with that - especially at this early juncture. The option of options approach is only non-breaking if the client specifies the options struct a particular way anyway(using the `...` syntax).

>  I feel like if we want to add more options they really should be part of either PutOptions or WriteOptions because we've decided that we'll have on "Options" per operation.

My understanding was that we settled on an options struct per logical function that the db is performing. In this case `PutOptions` covers specifying details about a row being added to the db, and `WriteOptions` specifies details about how a (batch of) row(s) should be written. From that lens specifying details about how a transaction should behave feels like it requires a separate options type.

> if there's a single-update transaction somehow it should be part of PutOptions

By that logic wouldn't it make sense to have a single flat options struct for `Db::put_with_options`?

Reply:
> By that logic wouldn't it make sense to have a single flat options struct for Db::put_with_options?

yes, the thing that comes into play here is that we allow users to use one method to both put & write a single row for convenience, but it's logically the same as `write(WriteBatch::new().put_with_options(..., PutOptions), WriteOptions)`

anyway, it seems like we're OK with what we have for now so we'll leave it as is. we can settle on the final API after transactions/checkpoints/etc... come in and we have more options we need to deal with